Update link to WASM ICU filters (#33895)
dotnet/docs: docs/core/extensions/globalization-icu.md

- <xref:System.String.Normalize(System.Text.NormalizationForm)?displayProperty=nameWithType> and <xref:System.String.IsNormalized(System.Text.NormalizationForm)?displayProperty=nameWithType> don't support the rarely used <xref:System.Text.NormalizationForm.FormKC> and <xref:System.Text.NormalizationForm.FormKD> forms.
- <xref:System.Globalization.RegionInfo.CurrencyNativeName?displayProperty=nameWithType> returns the same value as <xref:System.Globalization.RegionInfo.CurrencyEnglishName?displayProperty=nameWithType>.

> In addition, a list of supported locales is reduced and can be found on the [dotnet/icu repo](https://github.com/dotnet/icu/blob/dotnet/main/icu-filters/icudt_wasm.json#L7-L195).

Reviewer: ```suggestion
In addition, fewer locales are supported. The supported list can be found in the [dotnet/icu repo](https://github.com/dotnet/icu/blob/dotnet/main/icu-filters/icudt_wasm.json#L7-L195).
```

> In addition, a list of supported locales is reduced and can be found on the [dotnet/icu repo](https://github.com/dotnet/icu/blob/5667bcd34fd5edab2ce2e01fdb9533306b8fcf61/icu-filters/icudt_wasm.json#L7-L195).

Reviewer 2: Is it cleaner (better) to link to the repo/branch with `dotnet/main`? 

```suggestion
In addition, a list of supported locales is reduced and can be found on the [dotnet/icu repo](https://github.com/dotnet/icu/blob/dotnet/main/icu-filters/icudt_wasm.json#L7-L195).
```
